### ISSUE TITLE
DAE files import with many duplicate materials

### ISSUE THREAD

**Issue by SHFRH** [Frooxius/NeosPublic#668]
Having just read about the nice new "Deduplicate Instances" import option I wonder if a  "Deduplicate materials" option might also be possible?

I recent fix for importing DAE files from the 3Dwarehouse made millions of models usable but they still import with many meshes and duplicate materials. Baking the meshes solves the first issue but the second remains a challenge.

The simple arbitrary example model attached (featured on the front page of the warehouse today) shows this problem but some models import with hundreds of duplicate materials. Thanks, ^H^

[vespa2.zip](https://github.com/Frooxius/NeosPublic/files/4762221/vespa2.zip)

**Comment by Frooxius** [comment]
Hmm, I could run explicitly the material deduplication optimization after the model finishes importing (it's done to deduplicate same materials in the world, which is ran automatically during saving and such). 

You can also do this manually by spawning the AssetOptimizationWizard and using the "Deduplicate Materials" on the hierarchy. 

However I can't replicate the issue with the model you provided. How many materials do you get after import? I get 15 and each of them is unique.

![VespaMaterials](https://user-images.githubusercontent.com/8838625/84339697-47bdeb00-ab9f-11ea-8ed7-2746ec5c926c.jpg)

**Comment by SHFRH** [comment]
Hmm, perhaps I don't know how this works (surprise!)
A more typical example below list 217 materials in the inspector (lots of duplicate names) but when I do your trick to extract all the materials I get 16 orbs. Does that mean I should ignore long lists of materials (don't need optimizing by deduplicating i.e. for Quest)? Thanks

[Apollo+-+Lunar+Module.zip](https://github.com/Frooxius/NeosPublic/files/4762454/Apollo%2B-%2BLunar%2BModule.zip)

**Comment by Frooxius** [comment]
Oh I see what do you mean. The actual number of materials is pretty low, but the mesh itself has hundreds of submeshes, that use the same small set of materials.

I've tried the Apollo Lunar Module you provided. If you enable the "Deduplicate Instances" and "Optimize Model/Scene" options in advanced settings it should get rid of it. In case of the Apollo model it reduced it to just 10 submeshes on each node from the hundreds.

**Comment by SHFRH** [comment]
Sounds great - will definitely give that a try. Thanks
(the other day we had one with more than a thousand materials and just trashed it saying 'to hard')

**Comment by ProbablePrime** [comment]
I took a look at this model after chatting with Hamish about this in a world. At first I was very confused and thought that we were all talking about separate items:
- Slots
- Meshes
- Sub Meshes
- Material Slots on Meshes.

Then I did some reading of some historic discord conversations about how SubMeshes work and understood.

I tested the model with the new optimisation options: 

Before on the left, After on the right:
![20200614033756_1](https://user-images.githubusercontent.com/8791132/84591735-a1b7ee00-adf5-11ea-9d1d-63b6b27c5704.jpg)

It looks like these options do what we need for DAE files and as such this issue can be closed.
